solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -318,13 +405,24 @@ public long getChecksum() throws IOException {
     @Override
     public void writeByte(byte b) throws IOException {
       backing.writeByte(b);
-      sizeWriter.apply(1, name);
+      long postBytesWritten = getBytesWritten(backing);
+      sizeWriter.apply(1, postBytesWritten - lastBytesWritten, name);
+      lastBytesWritten = postBytesWritten;
+    }
+
+    private long getBytesWritten(IndexOutput out) {
+      if (backing instanceof CompressingDirectory.SizeReportingIndexOutput) {
+         return ((CompressingDirectory.SizeReportingIndexOutput) backing).getBytesWritten();
+      }
```

Comment:
Can probably leave this as-is, but it would have been nice to do the `instanceof` check in the ctor. Except that we'd have to keep the raw `sizeWriter` around too, it occurred to me that it would have been possible directly incorporate the `getBytesWritten()` stuff into a shimmed `sizeWriter`. Not totally happy with that though, so unless something strikes your fancy I think just leave this as-is.